lib/src/arithmetic/values/fixed_point_value.dart

```diff
@@ -0,0 +1,257 @@
+    }
+  }
+
+  /// Expands the bit width of integer and fractional parts.
+  LogicValue expandWidth({required bool sign, int m = 0, int n = 0}) {
+    if ((m > 0) & (m < this.m)) {
+      throw RohdHclException('Integer width is larger than input.');
+    }
+    if ((n > 0) & (n < this.n)) {
+      throw RohdHclException('Fraction width is larger than input.');
+    }
+    var newValue = value;
+    if (m >= this.m) {
+      if (signed) {
+        newValue = newValue.signExtend(newValue.width + m - this.m);
+      } else {
+        newValue = newValue.zeroExtend(newValue.width + m - this.m);
+        if (sign) {
+          newValue = newValue.zeroExtend(newValue.width + 1);
+        }
+      }
+    }
+    if (n > this.n) {
+      newValue =
+          newValue.reversed.zeroExtend(newValue.width + n - this.n).reversed;
+    }
+    return newValue;
+  }
+
+  /// Returns a negative integer if `this` less than [other],
+  /// a positive integer if `this` greater than [other],
+  /// and zero if `this` and [other] are equal.
+  @override
+  int compareTo(Object other) {
+    if (other is! FixedPointValue) {
+      throw RohdHclException('Input must be of type FixedPointValue');
+    }
+    if (!value.isValid | !other.value.isValid) {
+      throw RohdHclException('Inputs must be valid.');
+    }
+    final s = signed | other.signed;
+    final m = max(this.m, other.m);
+    final n = max(this.n, other.n);
+    final val1 = expandWidth(sign: s, m: m, n: n);
+    final val2 = other.expandWidth(sign: s, m: m, n: n);
+    final comp = val1.compareTo(val2);
+    if (comp == 0) {
+      return comp;
+    } else if (!isNegative() & !other.isNegative()) {
+      return comp;
+    } else if (!isNegative() & other.isNegative()) {
+      return 1;
+    } else if (isNegative() & !other.isNegative()) {
+      return -1;
+    } else {
+      return -comp;
+    }
+  }
+
+  /// Equal-to operation that returns a LogicValue.
+  LogicValue eq(FixedPointValue other) =>
+      compareTo(other) == 0 ? LogicValue.one : LogicValue.zero;
+
+  /// Not equal-to operation that returns a LogicValue.
+  LogicValue neq(FixedPointValue other) =>
+      compareTo(other) != 0 ? LogicValue.one : LogicValue.zero;
+
+  /// Less-than operation that returns a LogicValue.
+  LogicValue operator <(FixedPointValue other) =>
+      compareTo(other) < 0 ? LogicValue.one : LogicValue.zero;
+
+  /// Less-than operation that returns a LogicValue.
+  LogicValue operator <=(FixedPointValue other) =>
+      compareTo(other) <= 0 ? LogicValue.one : LogicValue.zero;
+
+  /// Less-than operation that returns a LogicValue.
+  LogicValue operator >(FixedPointValue other) =>
+      compareTo(other) > 0 ? LogicValue.one : LogicValue.zero;
+
+  /// Less-than operation that returns a LogicValue.
+  LogicValue operator >=(FixedPointValue other) =>
+      compareTo(other) >= 0 ? LogicValue.one : LogicValue.zero;
+
+  @override
+  int get hashCode => _hashCode;
+
+  int get _hashCode =>
```

Comment:
this seems unnecessary to have the private copy, you could just put `int get hashCode =>  value.hashCode ^ signed.hashCode ^ m.hashCode ^ n.hashCode;`